### ISSUE TITLE
fill prop not working for svg file!

### ISSUE THREAD

**Issue by cmcodes1** [kristerkari/react-native-svg-transformer#105]
Hey! @kristerkari I was looking for a way to change the color of the svg icon ( just like in #4 ).
Going through the docs, so far I've tried this:

Imported the svg file in the project.
```import Shop from '../Components/Assets/shop.svg';```

Used the svg file.
```<Shop width={'100%'} height={'30%'} fill={'red'} />```

Created a ```.svgrrc``` (no extension) file in the root of the project containing:
```
{
  "replaceAttrValues": {
    "#748A9D": "{props.fill}"
  }
}
```

My project dependencies:
    "react-native": "0.63.2",
    "react-native-svg": "^12.1.0",
    "react-native-svg-transformer": "^0.14.3",
    "semver": "^7.3.2"
    "@svgr/core": "^5.4.0",
    "path-dirname": "^1.0.2",

Am I missing something?

**Comment by kristerkari** [comment]
That looks correct to me. Anu chance that you can upload your project somewhere, so that I can have a look at it?

**Comment by burhan3759** [comment]
Same here, I have the same issue. 

it only work with "#000"

**Comment by kristerkari** [comment]
@burhan3759 please provide some example of your problem, otherwise it's impossible to help you.

**Comment by burhan3759** [comment]
Sure, sorry for giving with lack of info. I had the exact similar problem with @cmcodes1 

1. When I change the fill in "ArrowLeftIcon" to some color it does not work
2. when i change the fill in SVG icon to #000000, then it works
3. I already added the #ffffff in svgrrc

Here is my svgrrc:

```
{
  "replaceAttrValues": {
    "#000": "{props.fill}",
    "#000000": "{props.fill}",
    "#ffffff": "{props.fill}",
  }
}
```

Here is my SVG icon:

``` 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1"  width="24" height="24" viewBox="0 0 24 24">
   <path fill="#ffffff" d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z" />
</svg> 
```

Here is how i use the icon:
```
<ArrowLeftIcon height="25" width="25" fill="#4D4D4D" />
```


Version of Libraries:
"react-native": "0.62.2",
"react-native-svg": "^12.1.0",
"react-native-svg-transformer": "^0.14.3",

**Comment by mofmn** [comment]
I have a similar issue, mine works sometimes and i cant explain why, and i cant reliably control the colour. it seems to me like its an issues with the .svgrrc file being loaded, as i had a universal transformation
 `{
  "replaceAttrValues": {
    "black": "red"
  }
}`

And one svg was following this rule while it no longer existed, i tried to restart everything. Perhaps did not run a clean build. but this rule seems to persist along with other rules created after. its a nightmare, its very hard to get consistent reproducibility. Otherwise i have the same set up as the others

**Comment by jmlavoier** [comment]
I had the same issue in Android, I'm not using expo. The resolution for me was

- 1. I've stopped the metro server and closed the emulator
- 2. Cleaned the build project `$ cd android`, `$ ./gradlew clean`
- 3. Restarted the  metro server with reset-cache `$ yarn start --reset-cache`
- 4. Run android again `$ yarn android`

After I've done this, I didn't have to clean the build anymore. I just had to `reset-cache` when I had to add more configs.

**Comment by sypl** [comment]
Also having the same problem, this config:
```
{
  "replaceAttrValues": {
    "#02024D": "{props.fill}"
  }
}
```

... does not work with this
```
import Add from '~/assets/svg-icons/Add.svg'

...

<Add height={50} width={50} fill="red" />
```

I'm having to replace "#02024D" with "@fill" in svg file and then
```
{
  "replaceAttrValues": {
    "@fill": "{props.fill}"
  }
}
```

**Comment by KPS250** [comment]
Could not change color for svg  with config file 

.svgrrc
`{
  "replaceAttrValues": {
    "#000": "{props.fill}",
    "#000000": "{props.fill}",
  }
}`

SVG Use: 
`
import Ribbon from '../../../assets/icons/ribbon.svg';
<Ribbon fill="#4D4D4D" height="150" width="150" />

`


SVG file
`<svg xmlns="http://www.w3.org/2000/svg" width="100" height="30" viewBox="0 0 100 30">
    <g fill="none" fill-rule="evenodd">
        <g>
            <g>
                <path fill="#000" d="M95.621 2c-11.127 2.514-31.018 9.855-31.51 3.218-.493-6.637.787 11.364 1.378 18.705.492 7.24 20.187-.302 31.511-2.313l-11.423-7.341L95.621 2z" transform="translate(-46 -151) translate(47 152)"/>
                <g>
                    <path stroke="#000" d="M95.621 2c-11.127 2.514-31.018 9.855-31.51 3.218-.493-6.637.787 11.364 1.378 18.705.492 7.24 20.187-.302 31.511-2.313l-11.423-7.341L95.621 2z" transform="translate(-46 -151) translate(47 152)"/>
                    <path fill="#000" d="M2.087 27.625C28.323 30.838 77.913 12.066 80 21.402L78.708 3.533C70.46-8.212 28.422 13.572 0 7.85c.1.803 14.907 10.841 14.907 10.841l-12.82 8.934z" transform="translate(-46 -151) translate(47 152)"/>
                </g>
                <path fill="#000" d="M78.872 7c.204 2.54-3.972 2.642-6.723 1.93-2.75-.609-4.11-1.625-7.268 0l1.08 14.735c.509-2.236 4.558-.711 7.309 0 2.75.61 6.927.61 6.723-1.93L78.873 7z" transform="translate(-46 -151) translate(47 152)"/>
                <path stroke="#F8F6F4" d="M2.087 27.625C28.323 30.838 77.913 12.066 80 21.402L78.708 3.533C70.46-8.212 28.422 13.572 0 7.85c.1.803 14.907 10.841 14.907 10.841l-12.82 8.934z" transform="translate(-46 -151) translate(47 152)"/>
            </g>
        </g>
    </g>
</svg>
`

**Comment by mhammerc** [comment]
@KPS250 
I exported my SVG from Figma and it set the property `fill` inside the `<svg>` to `none`.

`<svg fill="none">`

When that happen, every `<path>` (which are inside `<svg>`) are not getting colored, even if they are `<path fill="#000000">`.

To fix the issue, I had to change `fill="none"` to `fill="#000000"` inside the `<svg>` element.

**Comment by KPS250** [comment]
@mhammerc yeah I found this later. Thanks for the help

**Comment by Wituz** [comment]
In Expo, it works after I clear the cache. 

`expo r -c`

It does not seem like the hot reload pick up changes in .svgrrc, so I have to clear the cache every time I make a change with the command above.

**Comment by SohaibIbneAli** [comment]
try all solutions but no one is working.

**Comment by mohammadameer** [comment]
> In Expo, it works after I clear the cache.
> 
> `expo r -c`
> 
> It does not seem like the hot reload pick up changes in .svgrrc, so I have to clear the cache every time I make a change with the command above.

@Wintus thank you very much this worked for me

**Comment by Alonew0lfxx** [comment]
> @KPS250
> I exported my SVG from Figma and it set the property `fill` inside the `<svg>` to `none`.
> 
> ```
> <svg fill="none">
> ```
> 
> When that happen, every `<path>` (which are inside `<svg>`) are not getting colored, even if they are `<path fill="#000000">`.
> 
> To fix the issue, I had to change `fill="none"` to `fill="#000000"` from the `<svg>` element.
> 
> **TLDR** Ensure `<svg>` elements have `fill="#000000"` and nothing else

Also I fixed my problem by removing fill="white" in the `<path></path>` line
and changing the `<svg fill="none">` to `<svg fill="#000000">` also I restarted the metro server and actual app

Example:
Old
```
<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M17.775 9...." fill="white"/>
</svg>
```
New
```
<svg width="24" height="24" viewBox="0 0 24 24" fill="#000000" xmlns="http://www.w3.org/2000/svg">
<path d="M17.775 9...."/>
</svg>
```

**Comment by ivanjuliant** [comment]
> In Expo, it works after I clear the cache.
> 
> `expo r -c`
> 
> It does not seem like the hot reload pick up changes in .svgrrc, so I have to clear the cache every time I make a change with the command above.

this working for me , thank you.

**Comment by henrhie** [comment]
> try all solutions but no one is working.

try removing all 'fill=...' from paths only and leave fill='none' in main svg tag

**Comment by DevYuns** [comment]
> > try all solutions but no one is working.
> 
> try removing all 'fill=...' from paths only and leave fill='none' in main svg tag

It works for me. Thanks!

**Comment by stanislav-sidorov** [comment]
in SVG children add fill="currentColor", then you can style

**Comment by uwemneku** [comment]
> @KPS250 I exported my SVG from Figma and it set the property `fill` inside the `<svg>` to `none`.
> 
> ```
> <svg fill="none">
> ```
> 
> When that happen, every `<path>` (which are inside `<svg>`) are not getting colored, even if they are `<path fill="#000000">`.
> 
> To fix the issue, I had to change `fill="none"` to `fill="#000000"` from the `<svg>` element.
> 
> **TLDR** Ensure `<svg>` elements have `fill="#000000"` and nothing else

For bare react-native project
adding a .svgrrc file to the root folder with the code
`{
  "replaceAttrValues": {
    "#000": "{props.fill}"
  }
}` worked for me

**Comment by nabil-atechy** [comment]
> @KPS250 I exported my SVG from Figma and it set the property `fill` inside the `<svg>` to `none`.
> 
> ```
> <svg fill="none">
> ```
> 
> When that happen, every `<path>` (which are inside `<svg>`) are not getting colored, even if they are `<path fill="#000000">`.
> 
> To fix the issue, I had to change `fill="none"` to `fill="#000000"` from the `<svg>` element.
> 
> **TLDR** Ensure `<svg>` elements have `fill="#000000"` and nothing else

thanks you save my day

**Comment by ritesh985** [comment]
> fill

It worked. Thanks!

**Comment by ritesh985** [comment]
> > try all solutions but no one is working.
> 
> try removing all 'fill=...' from paths only and leave fill='none' in main svg tag

It worked. Thanks!

**Comment by rajAmukhliS** [comment]
### fill prop not working for svg file!

### Solution:
I solved this issue 
**follow me to solve this** 

**Steps**

1. Open the svg in editor as you can see in picture below
![image](https://user-images.githubusercontent.com/71121356/149316269-fe2ff0ac-9546-467b-baf2-dabdcb7c40a3.png)

2. Remove all fill prop from paths only (**only from path tags**) , And ensure that svg tag's prop **fill ='null'** as you can see in picture below
![image](https://user-images.githubusercontent.com/71121356/149316726-e59f4d3f-1ad2-48fc-8c26-56e1a5098128.png)

3. Save that changes in svg file
4. Import that svg where you want to use 
5. Pass _fill={`Color of your Choice`}_ prop to that svg e.g `<NoStoreFoundSvg fill={`red`} />`
6. Save all files and you will get your desired result
Happy Coding!

**Comment by nabilcambiaso** [comment]
i'm using react-native-svg-transformer without using react-native-svg which i found very heavy in term of size,
so i can resize and change the stroke color also the fill color, but just instead of passing a fill prop, just pass color as seen below, it works perfectly

```
import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import { StatusBar } from 'expo-status-bar';
import Logo from "../../assets/profile.svg";

function FirstScreen(props) {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={() => { props.navigation.navigate('SecondScreen'); }}
      >
        <Text>Welcome</Text>
        <View style={{ aspectRatio: 1,justifyContent:"center",alignItems:"center", backgroundColor: 'blue',width:200,height:200 }}>
        <Logo color="white" stroke="black" height={50} width={50} />
        </View>
      </TouchableOpacity>
      <StatusBar style="auto" />
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
export default FirstScreen;
```


the svg code
`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 64 64"><title>profile</title><g fill="currentColor" class="nc-icon-wrapper"><path d="M38,37H26A19.021,19.021,0,0,0,7,56a1,1,0,0,0,.594.914C7.97,57.081,16.961,61,32,61s24.03-3.919,24.406-4.086A1,1,0,0,0,57,56,19.021,19.021,0,0,0,38,37Z"></path><path data-color="color-2" d="M32,32c8.013,0,14-8.412,14-15.933a14,14,0,1,0-28,0C18,23.588,23.987,32,32,32Z"></path></g></svg>`

dependencies

```
  "dependencies": {
    "@expo/webpack-config": "~0.16.2",
    "@react-navigation/native": "^6.0.10",
    "@react-navigation/native-stack": "^6.6.2",
    "expo": "~45.0.0",
    "expo-font": "^10.1.0",
    "expo-status-bar": "~1.3.0",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-native": "0.68.2",
    "react-native-svg-transformer": "^1.0.0",
  },
```

**Comment by Rakxhit** [comment]
> > try all solutions but no one is working.
> 
> try removing all 'fill=...' from paths only and leave fill='none' in main svg tag

This is how i achieved.

no need of .svgrrc file 
in your svg file remove all fills then your svg will be invisible,
then you can use the fill prop and change the colors,

**Comment by YaBuLiang** [comment]
已解决 记事本或者编辑器打开svg文件 删除svg图片文件里面的所有"fill="xxx"" 这样fill属性才会生效 可以改为你想要的颜色

**Comment by lada** [comment]
I can confirm that the color replacements do not work if you have fill="none" on the svg tag.
To solve it, you can either remove the attribute manually or **enable svgo in your .svgrrc file**

```
{
  "replaceAttrValues": {
    "#000": "{props.fill}"
  },
  "svgo": true
}
```

The default SVGO config includes the [removeUselessStrokeAndFill](https://github.com/svg/svgo/blob/main/plugins/removeUselessStrokeAndFill.js) plugin which will remove the fill="none" from the svg tag. The color replacements then work.

**Comment by DanDoW** [comment]
@lada You just helped me!

**Comment by rgomezp** [comment]
Nothing I have tried has worked.

I'm using Expo and the cache clearing solution did not work for me (looks like outdated command).

```
<svg
   version="1.1"
   id="svg1"
   width="256"
   height="640"
   viewBox="0 0 256 640"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:svg="http://www.w3.org/2000/svg">
  <defs
     id="defs1" />
  <g
     id="layer7"
     style="display:inline">
    <path
      fill="#999999"
       style="display:inline;fill:#999999"
       d="m 182.88897,129.91589 c 0,0 2.15728,-4.07485 7.67032,-3.59546 5.51303,0.4794 7.91,3.11607 7.91,3.11607 0,0 5.51304,3.35576 6.71153,13.18334 1.19848,9.82759 1.43818,33.79731 1.43818,33.79731 0,0 -0.4794,5.27334 -3.83516,1.19849 -3.35576,-4.07486 -11.98486,-29.00337 -12.70395,-33.07822 -0.71909,-4.07485 -3.11606,-11.50547 -7.19092,-14.62153 z"
       id="path8" />
    <path
      fill="#999999"
       style="display:inline;fill:#999999"
       d="m 182.88897,129.91589 c 0,0 2.15728,-4.07485 7.67032,-3.59546 5.51303,0.4794 7.91,3.11607 7.91,3.11607 0,0 5.51304,3.35576 6.71153,13.18334 1.19848,9.82759 1.43818,33.79731 1.43818,33.79731 0,0 -0.4794,5.27334 -3.83516,1.19849 -3.35576,-4.07486 -11.98486,-29.00337 -12.70395,-33.07822 -0.71909,-4.07485 -3.11606,-11.50547 -7.19092,-14.62153 z"
       id="path33"
       transform="matrix(-1.0093018,0,0,1,257.44455,-0.33898305)" />
  </g>
</svg>
```

I created the SVG with inkscape so maybe has something to do with it?

**Comment by faizanali1192** [comment]
Just removed the fill prop inside path and give path="none" in the parent svg tag like this.
```
<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
// Other svg path ...
</svg>

```
If you want to give the default color to svg or want to fill prop working as well you can give the default color in the fill prop
```
<svg width="20" height="20" viewBox="0 0 20 20" fill="#5B5B5F" xmlns="http://www.w3.org/2000/svg">
</svg>

```

**Comment by matinzd** [comment]
For those who are looking for setting a default value you can do this:

```
{
  "replaceAttrValues": {
    "#000": "{props?.fill ?? '#000'}"
  }
}
```

In this case if you don't pass fill, color will still be passed to the transformer and the default value will apply.

**Comment by CR4ZED** [comment]
> `expo r -c`

thanks for sharing

**Comment by ironidevi** [comment]
> Sure, sorry for giving with lack of info. I had the exact similar problem with [@cmcodes1](https://github.com/cmcodes1)
> 
> 1. When I change the fill in "ArrowLeftIcon" to some color it does not work
> 2. when i change the fill in SVG icon to #000000, then it works
> 3. I already added the #ffffff in svgrrc
> 
> Here is my svgrrc:
> 
> ```
> {
>   "replaceAttrValues": {
>     "#000": "{props.fill}",
>     "#000000": "{props.fill}",
>     "#ffffff": "{props.fill}",
>   }
> }
> ```
> 
> Here is my SVG icon:
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
> <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1"  width="24" height="24" viewBox="0 0 24 24">
>    <path fill="#ffffff" d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z" />
> </svg> 
> ```
> 
> Here is how i use the icon:
> 
> ```
> <ArrowLeftIcon height="25" width="25" fill="#4D4D4D" />
> ```
> 
> Version of Libraries: "react-native": "0.62.2", "react-native-svg": "^12.1.0", "react-native-svg-transformer": "^0.14.3",

Work
